### PR TITLE
[ADD]New Features: 'Spectators' entry in the activity form

### PR DESCRIPTION
**Pull request by jananjoy**

### CODE REVIEW

File: project_event/models/project_task.py [hunk 1]
```diff
@@ -207,14 +210,30 @@ class Task(models.Model):
...
-            raise ValidationError(_('The Input value should be numeric'))
+    def onchange_spectators(self):
+        if self.spectators:
+            if self.spectators == '-':
+                return
+            try:
+                int(self.spectators)
+            except Exception as exception:
+                raise ValidationError(_('Spectators value should be numeric'))
+            if int(self.spectators) >= MIN_SPECTATORS_VALUES_LIMIT and\
```
Comment (rimbendhaou) [inline project_event/models/project_task.py]: better to set the spectators= int(self.spectators) in a variable and reuse it

File: project_event/models/project_task.py [hunk 2]
```diff
@@ -207,14 +210,30 @@ class Task(models.Model):
...
-            int(self.spectators)
-        except Exception as exception:
-            raise ValidationError(_('The Input value should be numeric'))
+    def onchange_spectators(self):
+        if self.spectators:
+            if self.spectators == '-':
+                return
+            try:
+                int(self.spectators)
+            except Exception as exception:
```
Comment (rimbendhaou) [inline project_event/models/project_task.py]: we dont need as exception if you will not use it

**Review by rimbendhaou — CHANGES_REQUESTED** [review]

File: project_event/models/project_task.py [hunk 3]
```diff
@@ -207,14 +210,30 @@ class Task(models.Model):
...
+        default='-',
+    )
 
     @api.onchange('spectators')
-    def check_numeric(self):
-        try:
-            int(self.spectators)
-        except Exception as exception:
-            raise ValidationError(_('The Input value should be numeric'))
+    def onchange_spectators(self):
```
Comment (rimbendhaou) [inline project_event/models/project_task.py:220]: add the unit test for this method

File: project_event/models/project_task.py [hunk 4]
```diff
@@ -207,14 +210,30 @@ class Task(models.Model):
...
+                return
+            try:
+                int(self.spectators)
+            except Exception as exception:
+                raise ValidationError(_('Spectators value should be numeric'))
+            if int(self.spectators) >= MIN_SPECTATORS_VALUES_LIMIT and\
+                    int(self.spectators) < MAX_SPECTATORS_VALUES_LIMIT:
+                self.spectators = int(self.spectators)
+            else:
+                raise ValidationError(_('Spectators value should be in the range['
```
Comment (rimbendhaou) [inline project_event/models/project_task.py:232]: Add the translation

**Review by rimbendhaou — CHANGES_REQUESTED** [review]

File: project_event/models/project_task.py (hunk 3 above)
Comment (jananjoy) [reply project_event/models/project_task.py:220]: @rimbendhaou  : added

File: project_event/models/project_task.py (hunk 4 above)
Comment (jananjoy) [reply project_event/models/project_task.py:232]: @rimbendhaou  Added

File: project_event/models/project_task.py (hunk 1 above)
Comment (jananjoy) [reply project_event/models/project_task.py]: @rimbendhaou :Ok

File: project_event/models/project_task.py (hunk 2 above)
Comment (jananjoy) [reply project_event/models/project_task.py]: @rimbendhaou  Ok

File: project_event/tests/test_project_event_task.py [hunk 5]
```diff
@@ -464,3 +464,9 @@ def test_200_compute_actual_total_time(self):
         self.assertEqual(diff, "00:00")
         diff = self.task_4.total_time
         self.assertEqual(diff, "02:00")
+
+    def test_201_onchange_spectators(self):
```
Comment (eilst) [inline project_event/tests/test_project_event_task.py]: You should also test the Exception

File: project_event/tests/test_project_event_task.py [hunk 6]
```diff
@@ -464,3 +464,15 @@ def test_200_compute_actual_total_time(self):
...
+        self.assertEqual(self.activity_3.spectators, '-')
+        self.activity_3.spectators = '123'
+        self.activity_3.onchange_spectators()
+        self.assertEqual(self.activity_3.spectators, '123')
+        self.activity_3.spectators = 'sfl'
+        self.activity_3.onchange_spectators()
+        self.assertEqual(self.activity_3.spectators, 'Spectators value should be numeric')
+        self.activity_3.spectators = '999999999999'
+        self.activity_3.onchange_spectators()
+        self.assertEqual(self.activity_3.spectators, 'Spectators value should be in the range[0-1000000]')
```
Comment (eilst) [inline project_event/tests/test_project_event_task.py]: with self.assertRaises(exceptions.YourExeptionType):
                     action:(write)

Comment (jananjoy) [reply project_event/tests/test_project_event_task.py]: @eilst :Added

**Pull request merged**